File: data/EXP01/python/2358/2358.output_2.py
# @lc app=leetcode id=2358 lang=python3
from typing import List

class Solution:
    def maximumGroups(self, grades: List[int]) -> int:
        grades.sort(reverse=True)  # Sort the grades in descending order
        
        groups = []
        current_group_sum = 0
        current_group_count = 0
        
        for grade in grades:
            if current_group_sum + grade >= current_group_count + 1:
                groups.append(current_group_sum)
                current_group_sum = grade
                current_group_count = 1
            else:
                current_group_sum += grade
                current_group_count += 1
                
        groups.append(current_group_sum)  # Add the sum of the last group
        
        return len(groups)

File: data/EXP01/golang_no_system_var/2358/2358.output_2.go
func maximumGroups(grades []int) int {
    sort.Sort(sort.Reverse(sort.IntSlice(grades))) // Sort the grades array in descending order

    maxGroups := 1 // At least one group can be formed
    totalSum := grades[0]
    totalCount := 1
    
    for i := 1; i < len(grades); i++ {
        totalSum += grades[i]
        totalCount++
        
        if totalSum >= grades[i]*(totalCount+1) {
            maxGroups++
        }
    }
    
    return maxGroups
}
